core/templates/dev/head/domain/utilities/BrowserCheckerService.js

```diff
@@ -40,9 +40,38 @@ oppia.factory('BrowserCheckerService', function() {
     }
   };
 
+  //www.sitepoint.com/community/t/detect-of-browser-is-safari/6761/2
```

Comment:
Two suggestions:
- The user's browser isn't going to change midway through the session. Let's just compute the browser type once and for all and store it in this service. Note also that there's some duplication with the stuff above; let's find a clean way to do all the checks and end up with a browser name.
- Along the lines suggested in the other PR, add a single BrowserCheckerService.supportsSpeechSynthesis() or similar. For now, just return true if it's Chrome and the speech synthesis checks are true.

This should keep things maintainable. Also, perhaps you could incorporate the stuff/changes in @DubeySandeep's PR? I think it's not a good idea to split this up between two different PRs by two different people, we should have just one person working on this. Maybe discuss with @DubeySandeep and figure out which of you wants to take it?

Reply:
- Yep, sounds good. I agree it's hacky but better to have one hack than two :)
- Ah, I see the difference, thank you for explaining. Let's have this service expose two methods, then: supportsSpeechSynthesis() and supportsAudioPlayback(). Safari would fail the latter.

Reply:
Sure I'll do that then.